tools/devservices_healthcheck.py

```diff
@@ -1,47 +1,183 @@
 from __future__ import annotations
 
+import argparse
 import os
 import subprocess
 import time
+from collections.abc import Sequence
+from typing import Callable
 
 
-def run_cmd(
-    args: list[str],
-    *,
-    retries: int = 3,
-    timeout: int = 5,
-) -> None:
-    for retry in range(1, retries + 1):
-        returncode = subprocess.call(args)
+class HealthcheckError(Exception):
+    pass
+
+
+class HealthCheck:
+    id: str
+    container_name: str
+    check_by_default: bool
+    check: Callable[[], None]
+    deps: list[str]
+    retries: int
+    timeout_secs: int
+
+    def __init__(
+        self, id, container_name, check_by_default, check=None, deps=None, retries=3, timeout_secs=5
+    ):
+        self.id = id
+        self.container_name = container_name
+        self.check_by_default = check_by_default
+        self.check = check
+        self.deps = deps or []
+        self.retries = retries
+        self.timeout_secs = timeout_secs
+
+    def check_container(self):
+        response = subprocess.run(
+            ("docker", "container", "inspect", "-f", "'{{.State.Status}}'", self.container_name),
```

Comment:
ah yep here it is -- the single quotes here are a bug.  you need these when you run in a shell but there's no shell here -- the process is being constructed directly